ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -139,6 +141,7 @@ byronTransition ByronPartialLedgerConfig{..}
                 eraStart
                 st@ByronLedgerState{..} =
       fmap cpuEpoch
+    . filterMaybe bumpsMajorProtocolVersion
     . latest
```

Comment:
`latest` returns only the proposal that will be adopted. `filterMaybe bumpsMajorProtocolVersion` then ignores it if it's irrelevant. We wouldn't want to filter first, because then we'll find the latest relevant one, but that's not the one that will be adopted!